### PR TITLE
test(Josh): add unit tests

### LINKED ISSUE #64: test(Josh): add unit tests
[`Josh`](https://github.com/josh-development/core/blob/main/src/lib/structures/Josh.ts) is in desperate need of appropriate tests.

### PR DESCRIPTION
**Pull request by dan-online**
resolves #64
- [x] 100% coverage
- [x] 100% line coverage
- [x] Fix map issues

### CODE REVIEW

File: src/lib/structures/Josh.ts [hunk 1]
```diff
@@ -700,40 +716,31 @@ export class Josh<StoredValue = unknown> {
...
    * @returns The value gotten or null.
    *
    * @example
    * ```javascript
    * await josh.set('key', 'value');
    *
    * await josh.get('key'); // 'value'
-   * ```
-   */
-  public async get(key: string): Promise<StoredValue | null>;
```
Comment (RealShadowNova) [inline src/lib/structures/Josh.ts:726]: why is this overload removed?

File: tests/lib/structures/Josh.test.ts [hunk 2]
```diff
@@ -11,7 +13,121 @@ describe('Josh', () => {
...
+        middlewares: [new AutoEnsureMiddleware({ defaultValue: { test: false } }) as unknown as Middleware<{ test: boolean }>]
+      });
+
+      expect(josh.middlewares.size).toBe(1);
+    });
+
+    test('GIVEN provider with invalid instance THEN emits warning', () => {
+      const spy = jest.spyOn(process, 'emitWarning').mockImplementation();
+      // @ts-expect-error - this is a test
+      const josh = new Josh({ name: 'test:instanceof', provider: new Map() });
```
Comment (RealShadowNova) [inline tests/lib/structures/Josh.test.ts:49]: what are the errors that come from this?

File: tests/lib/structures/Josh.test.ts [hunk 3]
```diff
@@ -21,4 +137,1083 @@ describe('Josh', () => {
...
+
+        // test('GIVEN josh w/ unique data THEN returns false', async () => {
+        //   await josh[Method.SetMany]([
+        //     ['firstKey', 'value'],
+        //     ['secondKey', 'not value']
+        //   ]);
+
+        //   const everyed = await josh[Method.Every]([], 'value');
+        //   expect(everyed).toBe(false);
+        // }); // For Nova
```
Comment (RealShadowNova) [inline tests/lib/structures/Josh.test.ts:456]: what is up with this

File: tests/lib/structures/Josh.test.ts [hunk 4]
```diff
@@ -21,4 +137,1083 @@ describe('Josh', () => {
...
+        const value = await josh[Method.Get]('test:math');
+
+        expect(value).toBe(1);
+      });
+    });
+
+    describe(Method.Partition, () => {
+      describe(Payload.Type.Hook, () => {
+        test('GIVEN josh w/o data THEN returns data w/o data', async () => {
+          const parted = await josh[Method.Partition]((value) => value === 'value');
```
Comment (RealShadowNova) [inline tests/lib/structures/Josh.test.ts:755]: could we have all these variables renamed to `result` or something else more consistent?

File: tests/lib/structures/Josh.test.ts [hunk 5]
```diff
@@ -21,4 +137,1083 @@ describe('Josh', () => {
...
+          await expect(some).rejects.toBeInstanceOf(JoshError);
+        });
+
+        // test('GIVEN josh w/ data THEN returns true', async () => {
+        //   await josh[Method.Set]('test:some', 'value');
+
+        //   const some = await josh[Method.Some]([], 'value');
+
+        //   expect(some).toBe(true);
+        // }); // For Nova
```
Comment (RealShadowNova) [inline tests/lib/structures/Josh.test.ts:1071]: what's up with this as well

**Review by RealShadowNova — CHANGES_REQUESTED** [review]

File: tests/lib/structures/Josh.test.ts (hunk 5 above)
Comment (dan-online) [reply tests/lib/structures/Josh.test.ts:1071]: These error with the getProperty issue where path is empty, some and every by value throws a missing data error

File: tests/lib/structures/Josh.test.ts (hunk 2 above)
Comment (dan-online) [reply tests/lib/structures/Josh.test.ts:49]: Map isn't a JoshProvider

File: tests/lib/structures/Josh.test.ts (hunk 3 above)
Comment (dan-online) [reply tests/lib/structures/Josh.test.ts:456]: See other comment

File: src/lib/structures/Josh.ts (hunk 1 above)
Comment (dan-online) [reply src/lib/structures/Josh.ts:726]: It isn't required afaik and it caused errors with more than 1 parameter

File: tests/lib/structures/Josh.test.ts (hunk 5 above)
Comment (RealShadowNova) [reply tests/lib/structures/Josh.test.ts]: Ok, can we create an issue for this on the utilities repository so a test can be added and then adapted in the providers repository when updating the dependency.

File: tests/lib/structures/Josh.test.ts (hunk 2 above)
Comment (RealShadowNova) [reply tests/lib/structures/Josh.test.ts:49]: we have the `@joshdb/map` dep so we could use that?

Comment (RealShadowNova) [reply tests/lib/structures/Josh.test.ts:49]: oh nvm

File: src/lib/structures/Josh.ts (hunk 1 above)
Comment (RealShadowNova) [reply src/lib/structures/Josh.ts:726]: It is because this forces the return value to `StoredValue`

Comment (dan-online) [reply src/lib/structures/Josh.ts:726]: Well when path is provided, the overload makes ts error that it only expected 1 argument, so is that something you can look at? I don't think I understand TS overloads enough in this area

Comment (RealShadowNova) [reply src/lib/structures/Josh.ts:726]: this should be fixed now

**Review by RealShadowNova — APPROVED** [review]

**Pull request merged**